01PITT_INST-ULS_1_5/js/custom.js

```diff
@@ -308,4 +317,89 @@ angular.module('getTempAddress', []).controller('prmRequestAfterController',func
 }).component('prmRequestAfter',{
   controller:'prmRequestAfterController'
 });
+
+
+//SMS Notifications User Preferences Interface
+angular.module('smsNotifications', []).controller('smsNotificationsController',function($scope,$http){
+	this.$onInit = function () {
+		$scope.smsNumber='';
+		//jwt is stored with double quotes?
+		var jwt=this.primoExplore.storageutil.sessionStorage.primoExploreJwt.slice(1, -1);
+
+		//SMS STATUS
+		$http({
+			method : "GET",
+			url : "https://dev-patron.libraries.pitt.edu/address/sms/?jwt="+jwt,
+			}).then(function mySuccess(response) {
+				//API responds with null if user has no SMS notification preference set
+				if (response.data == null){
+					$scope.isSmsSet = 'Not recieving notifications by SMS';  
+					$scope.smsNumber='';
+					$scope.smsStatus = false;
+     			}
+				else{
+					$scope.isSmsSet='Subscribed to SMS Notifications';
+					$scope.smsNumber = response.data;
+					$scope.smsStatus = true;
+				}
+			}, function myError(response) {
+				$scope.status = 'Error: Failed to determine SMS notification preferences';
+		});
+
+		//SUBSCRIBE
+		$scope.subscribe = function(){
+			if(this.smsNumber.match(/^[2-9]\d{2}-\d{3}-\d{4}$/) !== null){
+				$scope.waitingForResponse=true;
+				$http.put('https://dev-patron.libraries.pitt.edu/address/sms/?jwt='+jwt+'&sms='+this.smsNumber)
+					.then(function mySuccess(response) {
+						$scope.smsError = false;
+						$scope.waitingForResponse=false;
+						$scope.isSmsSet = 'SMS is ON';
+						$scope.smsStatus = true;
+ 					}, function myError(response) {
+						$scope.waitingForResponse=false;
+  						$scope.smsError = true;
+ 					});
+			}
+			else {
+				$scope.validationMessage = "Please use XXX-XXX-XXXX format";
+				$scope.validationColor = "pink";
+			}
+		}
+
+		//UNSUBSCRIBE
+		$scope.unsubscribe = function(){
+			$scope.waitingForResponse=true;
+			$http.delete('https://dev-patron.libraries.pitt.edu/address/sms/?jwt='+jwt)
+				.then(function mySuccess(response) {
+					$scope.smsError = false;
+					$scope.waitingForResponse=false;
+					$scope.isSmsSet = 'SMS is OFF';
+					$scope.smsStatus = false;
+				}, function myError(response) {
+					$scope.waitingForResponse=false;
+					$scope.smsError = true;
+			});
+		}
+	}
+ }).component('smsNotifications', {
+	require: {
+		primoExplore: '^primoExplore'
+	},
+	controller:'smsNotificationsController',
+	template: '\
+	<p ng-show="waitingForResponse">Processing your request</p>\
+	<p ng-show="!smsStatus && !waitingForResponse && !smsError"><strong>SMS notifications disabled</strong></p>\
```

Comment:
Suggested general strategy: bind scope variables/properties to data rather than boolean states

For example, instead of smsStatus and waitingForResponse and smsError, use a singular smsStatus for display, and change that message within the controller based on the users action.  Display smsStatus as a bound variable.

Reply:
A result of getting parts working one-by-one and not stepping back to take a broader view.  I've simplified as suggested using a single status in eeb177cc4bade6db410ae368533cb6165a2192db

Reply:
This looks quite simple and clear now!